inspect_wandb/models/hooks.py

```diff
@@ -37,8 +37,8 @@ def __init__(self):
 
     @override
     def enabled(self) -> bool:
-        self._load_settings()
-        assert self.settings is not None
+        # Always reload settings from scratch to pick up any runtime changes
+        self.settings = ModelsSettings.model_validate({})
```

Comment:
this seems fine for now - the idea with caching the settings was to improve performance, but that's probably a premature optimisation

cleanup suggestion: you could remove the `_load_settings` method and replace line 111 with `self.settings = ModelsSettings.model_validate(metadata_overrides)`